### PR TITLE
PrebidNative test cases

### PR DESCRIPTION
**Pull request by avermaiOS**
Added test cases of PrebidNative for bannerNative DFP & mopub.

### CODE REVIEW

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
+        assertEquals(PBS_CONFIG_ID_NATIVE_APPNEXUS, FieldUtils.readField(nativeUnit, "configId", true));
+        assertEquals(AdType.NATIVE, FieldUtils.readField(nativeUnit, "adType", true));
+    }
+
+    @Test
+    public void testNativeAdContextType() {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
```
Comment (anwzhang) [inline PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Not only check returned type is context type, also check the id fo the type is correct value.

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        assertNull(nativeUnit.params.getContextType());
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.CONTENT_CENTRIC);
+        assertNotNull(nativeUnit.params.getContextType());
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.CONTENT_CENTRIC);
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.SOCIAL_CENTRIC);
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.SOCIAL_CENTRIC);
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.PRODUCT);
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.PRODUCT);
+        nativeUnit.setContextType(NativeAdUnit.CONTEXT_TYPE.CUSTOM);
+        assertEquals(nativeUnit.params.getContextType(), NativeAdUnit.CONTEXT_TYPE.CUSTOM);
```
Comment (anwzhang) [inline PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Try to set a custom id then test the returned value.

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java [hunk 3]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    @Test
+    public void testNativeAdContextSubType() {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
+        assertNull(nativeUnit.params.getContextsubtype());
+        nativeUnit.setContextSubType(NativeAdUnit.CONTEXTSUBTYPE.GENERAL);
```
Comment (anwzhang) [inline PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: same as above.

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        nativeUnit.setContextSubType(NativeAdUnit.CONTEXTSUBTYPE.CHAT_IM);
+        assertEquals(nativeUnit.params.getContextsubtype(), NativeAdUnit.CONTEXTSUBTYPE.CHAT_IM);
+        nativeUnit.setContextSubType(NativeAdUnit.CONTEXTSUBTYPE.SELLING);
+        assertEquals(nativeUnit.params.getContextsubtype(), NativeAdUnit.CONTEXTSUBTYPE.SELLING);
+        nativeUnit.setContextSubType(NativeAdUnit.CONTEXTSUBTYPE.APPLICATION_STORE);
+        assertEquals(nativeUnit.params.getContextsubtype(), NativeAdUnit.CONTEXTSUBTYPE.APPLICATION_STORE);
+        nativeUnit.setContextSubType(NativeAdUnit.CONTEXTSUBTYPE.PRODUCT_REVIEW_SITES);
+        assertEquals(nativeUnit.params.getContextsubtype(), NativeAdUnit.CONTEXTSUBTYPE.PRODUCT_REVIEW_SITES);
+        nativeUnit.setContextSubType(NativeAdUnit.CONTEXTSUBTYPE.CUSTOM);
+        assertEquals(nativeUnit.params.getContextsubtype(), NativeAdUnit.CONTEXTSUBTYPE.CUSTOM);
```
Comment (anwzhang) [inline PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Try to set custom value and test.

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java [hunk 5]
```diff
@@ -0,0 +1,249 @@
...
+    }
+
+    @Test
+    public void testNativeAdPlacementType() {
+        NativeAdUnit nativeUnit = new NativeAdUnit(PBS_CONFIG_ID_NATIVE_APPNEXUS);
+        assertNotNull(nativeUnit);
+        assertNull(nativeUnit.params.getPlacementType());
```
Comment (anwzhang) [inline PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Same as above.

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java (hunk 1 above)
Comment (avermaiOS) [reply PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Done

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java (hunk 2 above)
Comment (avermaiOS) [reply PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Done

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java (hunk 4 above)
Comment (avermaiOS) [reply PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Done

File: PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java (hunk 5 above)
Comment (avermaiOS) [reply PrebidMobile/src/test/java/org/prebid/mobile/NativeAdUnitTest.java]: Done

**Comment by avermaiOS** [comment]
> Missing the test cases that creates a native ad unit, call load ad then validate the JSON object that’s created

Added

**Pull request merged**